code/game/objects/machinery/field_charger.dm

```diff
@@ -0,0 +1,292 @@
+ * Does not recharge itself wirelessly via APCs; must be wrenched down over a cable node or a terminal
+ * Is a child of /power/ because they have the code needed for connecting to cables and terminals already
+ */
+/obj/machinery/power/field_charger
+	name = "mobile field charger"
+	desc = "A study automated charger and storage rack for power cells and electronic devices. Has an internal battery that charges off the power grid when wrenched down."
+	icon = 'icons/obj/machines/vending.dmi'
+	icon_state = "lascharger"
+	interaction_flags = INTERACT_MACHINE_TGUI
+	resistance_flags = XENO_DAMAGEABLE|UNACIDABLE	//Don't want benos to cheese it by sneaking in and melting it
+	density = TRUE
+	anchored = FALSE
+	wrenchable = TRUE
+	drag_delay = FALSE
+	max_integrity = 1000
+	soft_armor = list(MELEE = 50, BULLET = 50, LASER = 50, ENERGY = 50, BOMB = 50, BIO = 100, FIRE = 50, ACID = 75)	//Harder to break with spit and gas
+	machine_max_charge = 100000	//Large power bank for when on the move
+	active_power_usage = 50000	//Looks like a lot but it's actually still pretty slow to re/charge
+	///List of references to power cells that are within this charger's contents list
+	var/list/obj/item/cell/stored_cells
+	///Like the contents list but stored in this manner so it doesn't have to be rebuilt unless an item is added/removed
+	var/list/stored_item_references
+	var/obj/machinery/power/apc/registered_apc
+
+/obj/machinery/power/field_charger/Initialize(mapload)
+	. = ..()
+	machine_current_charge = machine_max_charge
+
+/obj/machinery/power/field_charger/examine(mob/user)
+	. = ..()
+	. += "Charge: [span_bold("[PERCENT(machine_current_charge/machine_max_charge)]%")]"
+	. += "Integrity: [span_bold("[obj_integrity]/[max_integrity]")]"	//Not a percentage so players can see the exact values of both current and max integrity
+	. += "Has [span_bold("[LAZYLEN(stored_cells)]")] objects stored."
+
+/obj/machinery/power/field_charger/update_icon_state()
+	icon_state = initial(icon_state)
+	//Prevent divide by zero
+	if(!machine_current_charge)
+		icon_state += "_0"
+		return
+
+	switch(machine_current_charge / max(1, machine_max_charge))	//Max is in case it is 0
+		if(0 to 0.25)
+			icon_state += "_25"
+		if(0.25 to 0.5)
+			icon_state += "_50"
+		if(0.5 to 0.75)
+			icon_state += "_75"
+
+//Overriding this proc to change how the machine recharges itself because it does not draw from APCs; also why use_power() is not used
+/obj/machinery/power/field_charger/auto_use_power()
+	if(!powernet || machine_current_charge >= machine_max_charge)
+		return
+
+	//Find how much power the closest functioning APC connected to this grid has
+	var/apc_available_power = 0
+	//Check if there is an APC registered to this machine, if the breaker is on, if it's still on the same powernet, and if it has a cell with energy in it
+	if(!registered_apc || !registered_apc.operating || registered_apc?.powernet != powernet || registered_apc?.cell?.charge <= 0)
```

Comment:
Whut? This should draw from apcs and what I can see from reviewing on my phone that's what it does? What's going on here?

Reply:
Drawing from APCs is optional. It can accept power from a wire.